bundles/org.openhab.binding.philipsair/src/main/java/org/openhab/binding/philipsair/internal/connection/PhilipsAirAPIConnection.java

```diff
@@ -0,0 +1,252 @@
+                }
+            }
+
+            logger.debug("Philips Air Purifier device response: status = {}, content = '{}'", httpStatus, finalcontent);
+            switch (httpStatus) {
+                case OK_200:
+                    return finalcontent;
+                case BAD_REQUEST_400:
+                case UNAUTHORIZED_401:
+                case NOT_FOUND_404:
+                    logger.debug("Philips Air Purifier device responded with status code {}", httpStatus);
+                    throw new PhilipsAirAPIException(String.format("Error with status %d", httpStatus));
+                case TOO_MANY_REQUESTS_429:
+                    cooldownTimer = System.currentTimeMillis() + TimeUnit.MINUTES.toMillis(5);
+                default:
+                    logger.debug("Philips Air Purifier device responded with status code {}", httpStatus);
+                    throw new PhilipsAirAPIException(String.format("Error with status %d", httpStatus));
+            }
+        } catch (ExecutionException e) {
+            String errorMessage = e.getLocalizedMessage();
+            logger.trace("Exception occurred during execution: {}", errorMessage, e);
+            throw new PhilipsAirAPIException(errorMessage, e);
+        } catch (InterruptedException | TimeoutException e) {
+            logger.debug("Exception occurred during execution: {}", e.getLocalizedMessage(), e);
+            throw new PhilipsAirAPIException(e.getLocalizedMessage(), e);
+        } catch (Exception e) {
+            logger.warn("Unexpected exception occurred during execution: {}", e.getLocalizedMessage(), e);
+            throw new PhilipsAirAPIException(e.getLocalizedMessage(), e);
+        }
+    }
+
+    public @Nullable String exchangeKeys() throws PhilipsAirAPIException, InterruptedException {
+        if (this.cipher == null) {
+            return null;
+        }
+
+        String url = buildURL(KEY_URL, config.getHost());
+        String data = "{\"diffie\":\"" + this.cipher.getApow() + "\"}";
+        String encodedContent = getResponse(url, PUT, data, false);
+        JsonObject encodedJson = gson.fromJson(encodedContent, JsonObject.class);
+        String key = encodedJson.get("key").getAsString();
+        String hellman = encodedJson.get("hellman").getAsString();
+        String aesKey;
+        try {
+            aesKey = this.cipher.calculateKey(hellman, key);
+        } catch (GeneralSecurityException | TimeoutException | ExecutionException e) {
+            throw new PhilipsAirAPIException(e);
+        }
+
+        config.setKey(aesKey);
+        return aesKey;
+    }
+
+    public @Nullable PhilipsAirPurifierDataDTO sendCommand(String parameter, PhilipsAirPurifierWritableDataDTO value)
+            throws IllegalBlockSizeException, BadPaddingException, UnsupportedEncodingException, InvalidKeyException,
+            NoSuchAlgorithmException, NoSuchPaddingException, InvalidAlgorithmParameterException {

```

Comment:
You should wrap all of these exception in a `PhilipsAirAPIException` instead since I doubt the callers of this method would handle any of these exceptions differently.